src/stripe.common.ts

```diff
@@ -0,0 +1,498 @@
+Stripe.CryptoProvider = CryptoProvider;
+
+import DefaultPlatformFunctions = require('./platform/DefaultPlatformFunctions');
+Stripe._platformFunctions = new DefaultPlatformFunctions();
+
+import utils = require('./utils');
+
+function Stripe(
+  this: StripeObject,
+  key: string,
+  config: Record<string, unknown> = {}
+): void {
+  if (!(this instanceof Stripe)) {
+    return new (Stripe as any)(key, config);
+  }
+
+  const props = this._getPropsFromConfig(config);
+
+  Object.defineProperty(this, '_emitter', {
+    value: new EventEmitter(),
+    enumerable: false,
+    configurable: false,
+    writable: false,
+  });
+
+  this.VERSION = Stripe.PACKAGE_VERSION;
+
+  this.on = this._emitter.on.bind(this._emitter);
+  this.once = this._emitter.once.bind(this._emitter);
+  this.off = this._emitter.removeListener.bind(this._emitter);
+
+  if (
+    props.protocol &&
+    props.protocol !== 'https' &&
+    (!props.host || /\.stripe\.com$/.test(props.host))
+  ) {
+    throw new Error(
+      'The `https` protocol must be used when sending requests to `*.stripe.com`'
+    );
+  }
+
+  const agent = props.httpAgent || null;
+
+  this._api = {
+    auth: null,
+    host: props.host || DEFAULT_HOST,
+    port: props.port || DEFAULT_PORT,
+    protocol: props.protocol || 'https',
+    basePath: DEFAULT_BASE_PATH,
+    version: props.apiVersion || DEFAULT_API_VERSION,
+    timeout: utils.validateInteger('timeout', props.timeout, DEFAULT_TIMEOUT),
+    maxNetworkRetries: utils.validateInteger(
+      'maxNetworkRetries',
+      props.maxNetworkRetries,
+      0
+    ),
+    agent: agent,
+    httpClient: props.httpClient || Stripe.createNodeHttpClient(agent),
+    dev: false,
+    stripeAccount: props.stripeAccount || null,
+  };
+
+  const typescript = props.typescript || false;
+  if (typescript !== Stripe.USER_AGENT.typescript) {
+    // The mutation here is uncomfortable, but likely fastest;
+    // serializing the user agent involves shelling out to the system,
+    // and given some users may instantiate the library many times without switching between TS and non-TS,
+    // we only want to incur the performance hit when that actually happens.
+    Stripe.USER_AGENT.typescript = typescript;
+  }
+
+  if (props.appInfo) {
+    this._setAppInfo(props.appInfo);
+  }
+
+  this._prepResources();
+  this._setApiKey(key);
+
+  this.errors = _Error;
+  this.webhooks = require('./Webhooks');
+  this.webhooks._stripe = this;
+
+  this._prevRequestMetrics = [];
+  this._enableTelemetry = props.telemetry !== false;
+
+  this._requestSender = new RequestSender(
+    this,
+    Stripe.StripeResource.MAX_BUFFERED_REQUEST_METRICS
+  );
+  // Expose StripeResource on the instance too
+  // @ts-ignore
+  this.StripeResource = Stripe.StripeResource;
+
+  this._platformFunctions = Stripe._platformFunctions;
+}
+
+Stripe.errors = _Error;
+Stripe.webhooks = require('./Webhooks');
+
+Stripe.createNodeHttpClient = (agent: http.Agent): void => {
+  throw new Error(
```

Comment:
Should we also change which client we create by default for workers? It's a bit strange to make customers call a method when we already know the right thing to do.

Reply:
I'm thinking about the existing customers that use polyfils. If we keep calling createNodeHttpClient code that was working before will start breaking.

Reply:
That's a good point, I agree defaulting to `createFetchHttpClient` in worker environments is a better solution to avoid breaking customers. I wasn't sure if we wanted to quietly transition to fetch, or throw this error so customers know what changed and can provide their own implementation if they choose. If we call `createFetchHttpClient` by default, should we emit a warning too to let customers know their http client changed?